src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -2202,3 +2314,117 @@ def _validate_aci_location(norm_location):
     if norm_location not in aci_locations:
         raise CLIError('Azure Container Instance is not available at location "{}".'.format(norm_location) +
                        ' The available locations are "{}"'.format(','.join(aci_locations)))
+
+
+def osa_list(cmd, client, resource_group_name=None):
+    if resource_group_name:
+        managed_clusters = client.list_by_resource_group(resource_group_name)
+    else:
+        managed_clusters = client.list()
+    return _remove_osa_nulls(list(managed_clusters))
+
+
+def openshift_create(cmd, client, resource_group_name, name,  # pylint: disable=too-many-locals
+                     location=None,
+                     compute_vm_size="Standard_D4s_v3",
+                     compute_count=3,
+                     fqdn='',
+                     aad_client_app_id=None,
+                     aad_client_app_secret=None,
+                     aad_tenant_id=None,
+                     vnet_cidr="10.0.0.0/8",
+                     subnet_cidr="10.0.0.0/24",
+                     vnet_peer_id=None,
+                     tags=None,
+                     no_wait=False):
+
+    rg_location = _get_rg_location(cmd.cli_ctx, resource_group_name)
```

Comment:
This should be moved into the following `if` statement so the extra API call is only made if the location is not provided.